src/domain/lattice.ml

```diff
@@ -259,24 +256,36 @@ struct
     | (_, `Top) -> `Top
     | (`Bot, x) -> x
     | (x, `Bot) -> x
-    | (`Lifted x, `Lifted y) -> `Lifted (Base.join x y)
+    | (`Lifted x, `Lifted y) ->
+      try `Lifted (Base.join x y)
+      with TopValue -> `Top
```

Comment:
Earlier we would have raised here. Is swallowing this here the right choice? Also applies everywhere below.

Reply:
This shouldn't change any current behavior because currently we would have an uncaught `TopValue` exception, which I haven't seen anywhere.

The point of lifting a domain without a top to have a top is to be able to do joins without getting an exception. Otherwise every use case has to manually reimplement one of these domain lifters into the domain itself, like in https://github.com/goblint/analyzer/pull/1485#discussion_r1714983146.